#!/bin/bash
#PBS -j oe
#PBS -l nodes=1:ppn=8
#PBS -l mem=60Gb

#se or pe
unset READS2;
export READS1=`ls "$RNASEQ_SCRATCH"/data/"$RNASEQ_JOB_IDENT"/reads | tr "\t" "\n" | grep "$INPUT_FILE.*fastq" | head -n1;`;
echo $READS1 | grep "^.*_1.fastq" && export READS2=`echo $READS1 | sed 's/_1.fastq/_2.fastq/g'`

#if SE
if [ -z $READS2 ]; then
 STAR --runThreadN 8 --genomeDir $RNASEQ_ROOT/references/"$FASTA"_"$GTF"_"$RNASEQ_READ_LENGTH"bp/2pass --readFilesIn "$RNASEQ_SCRATCH"/data/"$RNASEQ_JOB_IDENT"/reads/"$READS1" --outFilterType BySJout --outFilterMultimapNmax 20 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --outSAMtype BAM Unsorted --outFileNamePrefix "$RNASEQ_SCRATCH"/data/"$RNASEQ_JOB_IDENT"/intermediate/"$READS1"_2ndpass && echo "$INPUT_FILE 2PASSALN SUCCESSFUL" >> "$RNASEQ_SCRATCH"/logs/"$RNASEQ_JOB_IDENT"/"$INPUT_FILE".log
else #if PE
 STAR --runThreadN 8 --genomeDir $RNASEQ_ROOT/references/"$FASTA"_"$GTF"_"$RNASEQ_READ_LENGTH"bp/2pass --readFilesIn "$RNASEQ_SCRATCH"/data/"$RNASEQ_JOB_IDENT"/reads/"$READS1" "$RNASEQ_SCRATCH"/data/"$RNASEQ_JOB_IDENT"/reads/"$READS2" --outFilterType BySJout --outFilterMultimapNmax 20 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000  --outSAMtype BAM Unsorted --alignMatesGapMax 1000000 --outFileNamePrefix "$RNASEQ_SCRATCH"/data/"$RNASEQ_JOB_IDENT"/intermediate/"$READS1"_2ndpass && echo "$INPUT_FILE 2PASSALN SUCCESSFUL" >> "$RNASEQ_SCRATCH"/logs/"$RNASEQ_JOB_IDENT"/"$INPUT_FILE".log
fi

exit 0
